### DISCUSSION TITLE
fr/datasets/detect/visdrone/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#13805] [Docs]
# fr/datasets/detect/visdrone/

Explore the VisDrone Dataset, a large-scale benchmark for drone-based image and video analysis with over 2.6 million annotations for objects like pedestrians and vehicles.

https://docs.ultralytics.com/fr/datasets/detect/visdrone/

**Comment by mariem158** [comment]
comment intégrer  visdataset sur google colab et l'intégrer dans mon code qui détecte seulement des véhicules par des drones situé verticalement au dessus de la route.

**Reply by pderrenger** [reply]
Bonjour,

Merci pour votre question ! Pour intégrer le dataset VisDrone dans Google Colab et l'utiliser pour détecter uniquement des véhicules capturés par des drones, voici les étapes à suivre :

1. **Configurer Google Colab**:
   - Assurez-vous d'avoir accès à un GPU pour accélérer l'entraînement. Allez dans `Runtime` > `Change runtime type` et sélectionnez `GPU`.

2. **Installer les dépendances**:
   - Installez les bibliothèques nécessaires, notamment `ultralytics` pour YOLOv8.

3. **Télécharger et préparer le dataset VisDrone**:
   - Utilisez le script de téléchargement et de conversion fourni dans le fichier `VisDrone.yaml`.

4. **Entraîner le modèle**:
   - Chargez un modèle pré-entraîné YOLOv8 et entraînez-le sur le dataset VisDrone en filtrant uniquement les classes de véhicules.

Voici un exemple de code pour chaque étape :

```python
# Installer les dépendances
!pip install ultralytics

# Importer les bibliothèques nécessaires
from ultralytics import YOLO
import os
from pathlib import Path

# Télécharger et préparer le dataset VisDrone
def download_visdrone():
    from ultralytics.utils.downloads import download

    def visdrone2yolo(dir):
        from PIL import Image
        from tqdm import tqdm

        def convert_box(size, box):
            dw = 1. / size[0]
            dh = 1. / size[1]
            return (box[0] + box[2] / 2) * dw, (box[1] + box[3] / 2) * dh, box[2] * dw, box[3] * dh

        (dir / 'labels').mkdir(parents=True, exist_ok=True)
        pbar = tqdm((dir / 'annotations').glob('*.txt'), desc=f'Converting {dir}')
        for f in pbar:
            img_size = Image.open((dir / 'images' / f.name).with_suffix('.jpg')).size
            lines = []
            with open(f, 'r') as file:
                for row in [x.split(',') for x in file.read().strip().splitlines()]:
                    if row[4] == '0':
                        continue
                    cls = int(row[5]) - 1
                    box = convert_box(img_size, tuple(map(int, row[:4])))
                    lines.append(f"{cls} {' '.join(f'{x:.6f}' for x in box)}\n")
                with open(str(f).replace(f'{os.sep}annotations{os.sep}', f'{os.sep}labels{os.sep}'), 'w') as fl:
                    fl.writelines(lines)

    dir = Path('../datasets/VisDrone')
    urls = [
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-train.zip',
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-val.zip',
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-test-dev.zip',
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-test-challenge.zip'
    ]
    download(urls, dir=dir, curl=True, threads=4)
    for d in ['VisDrone2019-DET-train', 'VisDrone2019-DET-val', 'VisDrone2019-DET-test-dev']:
        visdrone2yolo(dir / d)

download_visdrone()

# Entraîner le modèle YOLOv8 sur le dataset VisDrone
model = YOLO("yolov8n.pt")  # Charger un modèle pré-entraîné
results = model.train(data="VisDrone.yaml", epochs=100, imgsz=640)
```

Pour plus de détails sur la configuration du dataset et les options d'entraînement, vous pouvez consulter la documentation complète [ici](https://docs.ultralytics.com/datasets/detect/visdrone/).

N'hésitez pas à revenir vers nous si vous avez d'autres questions. Bonne chance avec votre projet ! 🚀

Cordialement,
L'équipe Ultralytics

**Reply by mariem158** [reply]
comment je peux utiliser les modèles de visdrone et faire l'appel à ce dataset dans mon code pour faire la détection des véhicules par mon drone?

**Reply by pderrenger** [reply]
Bonjour,

Merci pour votre question ! Pour utiliser le dataset VisDrone et détecter uniquement les véhicules capturés par des drones, voici comment procéder sur Google Colab :

1. **Configurer Google Colab**:
   - Assurez-vous d'avoir accès à un GPU pour accélérer l'entraînement. Allez dans `Runtime` > `Change runtime type` et sélectionnez `GPU`.

2. **Installer les dépendances**:
   - Installez les bibliothèques nécessaires, notamment `ultralytics` pour YOLOv8.

3. **Télécharger et préparer le dataset VisDrone**:
   - Utilisez le script de téléchargement et de conversion fourni dans le fichier `VisDrone.yaml`.

4. **Entraîner le modèle**:
   - Chargez un modèle pré-entraîné YOLOv8 et entraînez-le sur le dataset VisDrone en filtrant uniquement les classes de véhicules.

Voici un exemple de code pour chaque étape :

```python
# Installer les dépendances
!pip install ultralytics

# Importer les bibliothèques nécessaires
from ultralytics import YOLO
import os
from pathlib import Path

# Télécharger et préparer le dataset VisDrone
def download_visdrone():
    from ultralytics.utils.downloads import download

    def visdrone2yolo(dir):
        from PIL import Image
        from tqdm import tqdm

        def convert_box(size, box):
            dw = 1. / size[0]
            dh = 1. / size[1]
            return (box[0] + box[2] / 2) * dw, (box[1] + box[3] / 2) * dh, box[2] * dw, box[3] * dh

        (dir / 'labels').mkdir(parents=True, exist_ok=True)
        pbar = tqdm((dir / 'annotations').glob('*.txt'), desc=f'Converting {dir}')
        for f in pbar:
            img_size = Image.open((dir / 'images' / f.name).with_suffix('.jpg')).size
            lines = []
            with open(f, 'r') as file:
                for row in [x.split(',') for x in file.read().strip().splitlines()]:
                    if row[4] == '0':
                        continue
                    cls = int(row[5]) - 1
                    box = convert_box(img_size, tuple(map(int, row[:4])))
                    lines.append(f"{cls} {' '.join(f'{x:.6f}' for x in box)}\n")
                with open(str(f).replace(f'{os.sep}annotations{os.sep}', f'{os.sep}labels{os.sep}'), 'w') as fl:
                    fl.writelines(lines)

    dir = Path('../datasets/VisDrone')
    urls = [
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-train.zip',
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-val.zip',
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-test-dev.zip',
        'https://github.com/ultralytics/yolov5/releases/download/v1.0/VisDrone2019-DET-test-challenge.zip'
    ]
    download(urls, dir=dir, curl=True, threads=4)
    for d in ['VisDrone2019-DET-train', 'VisDrone2019-DET-val', 'VisDrone2019-DET-test-dev']:
        visdrone2yolo(dir / d)

download_visdrone()

# Entraîner le modèle YOLOv8 sur le dataset VisDrone
model = YOLO("yolov8n.pt")  # Charger un modèle pré-entraîné
results = model.train(data="VisDrone.yaml", epochs=100, imgsz=640)
```

Pour plus de détails sur la configuration du dataset et les options d'entraînement, vous pouvez consulter la documentation complète [ici](https://docs.ultralytics.com/datasets/detect/visdrone/).

N'hésitez pas à revenir vers nous si vous avez d'autres questions. Bonne chance avec votre projet ! 🚀

Cordialement,
L'équipe Ultralytics

**Comment by mslabdelhak29** [comment]
Hello can you offre a pretrained model i need it and i dont have source to entrain for 100 epochs

**Reply by glenn-jocher** [reply]
Hi, we don’t currently provide an official YOLO model pretrained specifically on VisDrone, so the usual workflow is to start from a generic checkpoint like `yolo11n.pt` and fine‑tune it on VisDrone; you can safely reduce `epochs` (e.g. 10–20) or train on a subset of the data in Colab to keep compute low, and the [YOLO train mode docs](https://docs.ultralytics.com/modes/train/) show how to adjust these arguments.
